### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by gilligan**
Revamp the whole text providing info on

- installation / setup via nixos module
- building via nix-build
- development environment via nix-shell

**Notes**

It is really nice that hydra is nowadays pretty easy to get up and running. Even more reason to provide the little necessary information directly at the source as part of the GitHub README.

Ideally i would like to extend this with a sub-section under Installation that briefly explains how to create a minimal project and jobset. In which case we could really consider someone "up and running" with only very few steps and explanation required.

But i would like to get feedback on the PR in its current state first and see if can be merged.

Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,63 @@
...
+$ nix-build
+```
+
+### Development Environment
+
+You can use the provided shell.nix to get a working development environment:
+```
+$ nix-shell
+$ ./bootstrap
+$ configurePhase # NOTE: note ./configure
```
Comment (grahamc) [inline README.md:49]:
```suggestion
$ configurePhase # NOTE: not ./configure
```

**Comment by grahamc** [comment]
Seems like a good start! Maybe some good stuff could be had from https://www.notion.so/grahamc/Hydra-Hacking-a455cc6c9ec1489d9144ed12c19d61e4 ?

**Comment by gilligan** [comment]
@grahamc sure, i sort of wanted to start out small but and see what people think but i'm happy to integrate a section under development that incorporates your info because it is definitely _very_ valuable for anyone who wants to do _actual_ work on hydra!

**Comment by grahamc** [comment]
Thanks!

**Pull request merged**